src/client/extension.ts

```diff
@@ -192,13 +222,41 @@ export async function activate(
     }
 
     _telemetry.sendTelemetryEvent("activated");
+    oneDSLoggerWrapper.getLogger().traceInfo("activated");
+}
 
+export function setupAuthFileWatcher() {
+    const watchPath = GetAuthProfileWatchPattern();
+    if (watchPath) {
+        const watcher = vscode.workspace.createFileSystemWatcher(watchPath);
+        watcher.onDidChange(() => handleOrgChange());
+        watcher.onDidCreate(() => handleOrgChange());
+        watcher.onDidDelete(() => handleOrgChange());
+    }
+}
+
+async function handleOrgChange() {
+    const cliContext = new CliAcquisitionContext(_context, _telemetry);
+    const cli = new CliAcquisition(cliContext);
+    const cliPath = await cli.ensureInstalled();
+    const pacContext = new PacWrapperContext(_context, _telemetry);
+    const interop = new PacInterop(pacContext, cliPath);
+    const pacWrapper = new PacWrapper(pacContext, interop);
+    const pacActiveOrg = await pacWrapper?.activeOrg();
+    if (pacActiveOrg && pacActiveOrg.Status === PAC_SUCCESS) {
+        const orgID = pacActiveOrg.Results.OrgId;
+        const artemisResponse = await fetchArtemisResponse(orgID, _telemetry);
+        if (artemisResponse) {
+            const { geoName } = artemisResponse[0];
+            oneDSLoggerWrapper.instantiate(geoName);
```

Comment:
Have you tested by changing geo. Is the data flowing to respective Aria clusters?

Reply:
On creating new project in Aria, data isn't flowing - so can't test.

Reply:
have created a tracking item: https://dynamicscrm.visualstudio.com/OneCRM/_workitems/edit/3821881
Will take it up here.